helios-services/src/main/java/com/spotify/helios/master/reaper/DeadAgentReaper.java

```diff
@@ -76,13 +75,12 @@ public DeadAgentReaper(final MasterModel masterModel,
   @Override
   void processItem(final String agent) {
     try {
-      final HostStatus hostStatus = masterModel.getHostStatus(agent);
-      if (hostStatus == null || hostStatus.getStatus() != HostStatus.Status.DOWN) {
-        // Host not found or host not DOWN -- nothing to do
+      if (masterModel.isHostUp(agent)) {
```

Comment:
Why change this logic at all?

AFAICT this is just doing the same thing as before, but adds a new method to do it.